src/components/MAPI/apps/AppList/utils.tsx

```diff
@@ -164,52 +123,63 @@ function compareAppCatalogs(
 export function mapAppCatalogsToFacets(
   appCatalogs: applicationv1alpha1.IAppCatalog[] = [],
   selectedAppCatalogs: Record<string, boolean> = {},
-  appCatalogErrors: Record<string, string> = {}
+  error: string = ''
 ): IFacetOption[] {
-  return appCatalogs.sort(compareAppCatalogs).map((appCatalog) => {
+  const sortedAppCatalogs = appCatalogs.sort(compareAppCatalogs);
+
+  const facetOptions: IFacetOption[] = [];
+  for (const appCatalog of sortedAppCatalogs) {
+    // TODO(axbarsan): Remove this once https://github.com/giantswarm/giantswarm/issues/17490 is done.
+    if (appCatalog.metadata.name === 'helm-stable') continue;
+
     const uiTitle = computeAppCatalogUITitle(appCatalog);
 
-    return {
+    facetOptions.push({
       value: appCatalog.metadata.name,
       checked: selectedAppCatalogs.hasOwnProperty(appCatalog.metadata.name),
       label: (
         <CatalogLabel
           catalogName={uiTitle}
           iconUrl={appCatalog.spec.logoURL}
           description={appCatalog.spec.description}
-          error={appCatalogErrors[appCatalog.metadata.name]}
+          error={error}
         />
       ),
-    };
-  });
+    });
+  }
+
+  return facetOptions;
 }
 
 /**
  * Search through the apps and find one that is in the
  * selected catalogs, and that matches the search query.
  * @param searchQuery
- * @param indexApps
+ * @param appCatalogEntries
+ * @param selectedAppCatalogs
  */
-export function filterAppCatalogIndexApps(
+export function filterAppCatalogEntries(
   searchQuery: string,
-  indexApps: IAppCatalogIndexApp[],
+  appCatalogEntries: applicationv1alpha1.IAppCatalogEntry[],
   selectedAppCatalogs: Record<string, boolean>
 ) {
   const normalizedQuery = searchQuery.trim().toLowerCase();
 
-  return indexApps.filter((indexApp) => {
-    if (!selectedAppCatalogs.hasOwnProperty(indexApp.catalogName)) return false;
+  return appCatalogEntries.filter((appCatalogEntry) => {
+    if (!selectedAppCatalogs.hasOwnProperty(appCatalogEntry.spec.catalog.name))
+      return false;
 
     if (normalizedQuery.length < 1) return true;
 
-    if (indexApp.versions.length < 1) return false;
-    const version = indexApp.versions[0];
-
     switch (true) {
-      case version.name.toLowerCase().includes(normalizedQuery):
-      case version.description.toLowerCase().includes(normalizedQuery):
-      case version.keywords &&
-        version.keywords.join(',').toLowerCase().includes(normalizedQuery):
+      case appCatalogEntry.spec.appName.toLowerCase().includes(normalizedQuery):
+      case appCatalogEntry.spec.chart.description
+        ?.toLowerCase()
+        .includes(normalizedQuery):
+      case appCatalogEntry.spec.chart.keywords
+        ?.join('')
```

Comment:
Without a separator between the keywords, would this potentially return unexpected matches (i.e. substrings created by the end of one keyword and the start of the next)?

Reply:
Great catch 👍